jwst/outlier_detection/utils.py

```diff
@@ -20,6 +22,67 @@
 _ONE_MB = 1 << 20
 
 
+class DiskAppendableArray:
+    """
+    Creates a temporary file to which to append data, in order to perform
+    timewise operations on a stack of input images without holding all of them
+    in memory.
+    
+    This class is purpose-built for the median computation during outlier detection 
+    and is not very flexible. It is assumed that each data array passed to `append`
+    represents the same spatial segment of the full dataset. It is also assumed that 
+    each data array passed to `append` represents only a single instant in time; 
+    the append operation will stack them along a new axis.
+    
+    The `read` operation is only capable of loading the full array back into memory.
+    When working with large datasets that do not fit in memory, the
+    required workflow is to create many DiskAppendableArray objects, each holding
+    a small spatial segment of the full dataset.
+    """
+    
+    def __init__(self, slice_shape, dtype='float32', tempdir="", filestem="data"):
+        """
+        Parameters
+        ----------
+        slice_shape : tuple
+            The shape of the spatial section of input data to be appended to the array.
+
+        dtype : str
+            The data type of the array.
+
+        tempdir : str
+            The directory in which to create the temporary files.
+
+        filestem : str
+            The stem of the temporary file name. The extension ".bits" will be added.
+        """
+        self._temp_dir = tempfile.TemporaryDirectory(dir=tempdir)
```

Comment:
It looks like this writes the tempdir to the current working directory by default - is that desired? And might it lead to more collisions in ops?  The output directory may not be the same as the current directory.

Reply:
I don't think it should lead to collisions because the temporary directory should be unique, even though yes by default it's a subdirectory within the cwd.  I have no idea what default would be desired in ops, but I believe (@braingram can tell me if this is not correct) `ModelLibrary` makes its temporary directory in the same way.

Reply:
Yeah ModelLibrary uses the same pattern for temporary files:
https://github.com/spacetelescope/stpipe/blob/ecd5d162be425c24db2498b34bcbaeccec4ac557/src/stpipe/library.py#L181
Although currently those aren't used in ops (as far as I'm aware the libraries should always be on_disk=False except for the one in resample which points to the already generated models in the output directory). Finding (and configuring) a suitable tmpdir is likely worth exploring if we don't disable all tempfiles for ops.

Reply:
Hmm, that seems like it might be a little problematic.  Why not just leave the default at None, so that the temporary directory is the system-dependent default?

Reply:
I can ask team Coffee about this at our 3:00 meeting

Reply:
@melanieclarke we talked about this with Hien, Eric, and Jesse.  It sounds like there's not really any "good" place to do file I/O on their end.  We confirmed though that right now all temporary files are written to the current working directory, including those made by the old (but currently operational) `ModelContainer` when `in_memory=False` as is the current outlier detection default.  Based on that conversation I think leaving this as the default is likely the safest thing to do.

Reply:
@braingram Sorry I missed this yesterday. The issue is that `/tmp` is mounted with the `noexec` flag:

```bash
cat << EOF > /tmp/myscript.sh
#!/usr/bin/env bash
echo "hello world"
EOF
chmod +x /tmp/myscript.sh
/tmp/myscript.sh
# result
-bash: /tmp/myscript.sh: Permission denied
```